soccer/src/soccer/coach/coach_node.cpp

```diff
@@ -0,0 +1,140 @@
+#include "coach_node.hpp"
+
+CoachNode::CoachNode(const rclcpp::NodeOptions& options) : Node("coach_node", options) {
+    coach_pub_ =
+        this->create_publisher<rj_msgs::msg::CoachStateInterpretation>("/strategy/coach", 10);
+    play_state_change_timer_ =
+        this->create_wall_timer(100ms, [this]() { check_for_play_state_change(); });
+
+    play_state_sub_ = this->create_subscription<rj_msgs::msg::PlayState>(
+        "/referee/play_state", 10,
+        [this](rj_msgs::msg::PlayState::SharedPtr msg) { play_state_callback(msg); });
+
+    world_state_sub_ = this->create_subscription<rj_msgs::msg::WorldState>(
+        "/vision_filter/world_state", 10,
+        [this](rj_msgs::msg::WorldState::SharedPtr msg) { world_state_callback(msg); });
+
+    // initialize all of the robot status subscriptions
+    // initialize our robots
+    for (int i = 0; i < kRobotsPerTeam; i++) {
+        robot_status_subs_[i] = this->create_subscription<rj_msgs::msg::RobotStatus>(
+            fmt::format("/radio/robot_status/robot_{}", i), 10,
+            [this](const rj_msgs::msg::RobotStatus::SharedPtr msg) {
+                ball_sense_callback(msg, true);
+            });
+    }
+
+    // initialize their robots
```

Comment:
we don't have robot status for their robots. robot_status 0-15 exists because we could have 16 uniquely identified robots in theory. but these would all be controlled by our team still (why would the other team send us ball_sense?)